### PR TITLE
[XEB] Sundry practical improvements

### PR DESCRIPTION
**Pull request by mpharrigan**
- Shuffle order of task execution
 - Save data as you generate it
 - Move more functionality up the xxOptions Hierarchy
 - More robust fitting
 - Return std.dev. of fit parameters
 - Bugfix in notebook when running on device
 - Allow benchmarking the same pair in different layers and keep them distinct
 - Add note about characterizing the same pair in different layers: they will not be kept distinct. This can be relaxed, but requires changing the datatype of XEBCalibrationResult and keeping around (layer_i, pair_i) indices which are likely more troublesome to work with than actual qubit pairs.

### CODE REVIEW

**Comment by mpharrigan** [comment]
@mrwojtek @balopat @dstrain115 PTAL

File: cirq/experiments/xeb_fitting.py [hunk 1]
```diff
@@ -453,22 +459,36 @@ def _fit_exponential_decay(cycle_depths: np.ndarray, fidelities: np.ndarray) ->
...
     slope, intercept, _, _, _ = scipy.stats.linregress(cycle_depths_pos, log_fidelities)
     layer_fid_0 = np.clip(np.exp(slope), 0, 1)
     a_0 = np.clip(np.exp(intercept), 0, 1)
 
-    (a, layer_fid), _ = scipy.optimize.curve_fit(
-        exponential_decay, cycle_depths, fidelities, p0=(a_0, layer_fid_0), bounds=((0, 0), (1, 1))
-    )
-    return a, layer_fid
+    try:
+        (a, layer_fid), pcov = scipy.optimize.curve_fit(
```
Comment (mrwojtek) [inline cirq/experiments/xeb_fitting.py:479]: Is it possible for this to optimization to fail (not converge?)

File: cirq/experiments/xeb_fitting.py [hunk 2]
```diff
@@ -531,13 +556,16 @@ def before_and_after_characterization(
...
-    joined_df['characterized_angles'] = [
-        characterization_result.final_params[pair] for pair, _, _ in joined_df.index
-    ]
+    # Remove (layer_i, pair_i) from the index. While we keep this for `fit_exponential_decays`
+    # so the same pair can be benchmarked in different contexts, the multi-pair characterization
+    # function only keys on the pair identity. This can be seen acutely by the
+    # `characterization_result.final_params` dictionary being keyed only by the pair.
+    joined_df = joined_df.reset_index().set_index('pair')
+
+    cr = characterization_result  # shorter name to prevent line wrapping
```
Comment (mrwojtek) [inline cirq/experiments/xeb_fitting.py:565]: I think this is less readable.

File: cirq/experiments/xeb_fitting.py [hunk 3]
```diff
@@ -453,22 +459,36 @@ def _fit_exponential_decay(cycle_depths: np.ndarray, fidelities: np.ndarray) ->
...
+        a_std: The standard deviation of the `a` parameter estimate.
+        layer_fid_std: The standard deviation of the `layer_fid` parameter estimate.
     """
     cycle_depths = np.asarray(cycle_depths)
     fidelities = np.asarray(fidelities)
 
     # Get initial guess by linear least squares with logarithm of model
     positives = fidelities > 0
+    if np.sum(positives) <= 1:
+        return 0, 0, np.inf, np.inf
```
Comment (mrwojtek) [inline cirq/experiments/xeb_fitting.py:471]: Could you comment on this here?

**Review by mrwojtek — APPROVED** [review]

File: cirq/experiments/xeb_fitting.py (hunk 1 above)
Comment (mpharrigan) [reply cirq/experiments/xeb_fitting.py:479]: yes, that's why the try except is being added. I couldn't figure out how to make a test dataset for a unit test to trigger the behavior, but the real device sometimes does :)

File: cirq/experiments/xeb_fitting.py (hunk 2 above)
Comment (mpharrigan) [reply cirq/experiments/xeb_fitting.py:565]: reverted

File: cirq/experiments/xeb_fitting.py (hunk 3 above)
Comment (mpharrigan) [reply cirq/experiments/xeb_fitting.py:471]: yes; this is a bit of numpy trickery. Added comment(s)

**Comment by balopat** [comment]
> Automerge cancelled: Merging is blocked (I don't understand why).

Hmmm....we really need to improve this path in automerge...
The logs say the same, haha: 

```
Cancelled automerge of PR#3871 ('[XEB] Sundry practical improvements'): Merging is blocked (I don't understand why).
```

**Review by tanujkhattar — APPROVED** [review]

**Comment by balopat** [comment]
> > Automerge cancelled: Merging is blocked (I don't understand why).
> 
> Hmmm....we really need to improve this path in automerge...
> The logs say the same, haha:
> 
> ```
> Cancelled automerge of PR#3871 ('[XEB] Sundry practical improvements'): Merging is blocked (I don't understand why).
> ```

It boils down to for some reason this json having `mergeable_state == 'blocked'`: https://api.github.com/repos/quantumlib/cirq/pulls/3871 -- currently it is `'clean'` - so this is a temporary condition. I think we should just retry from auto_merge...

<details>

```json
{
  "url": "https://api.github.com/repos/quantumlib/Cirq/pulls/3871",
  "id": 582332259,
  "node_id": "MDExOlB1bGxSZXF1ZXN0NTgyMzMyMjU5",
  "html_url": "https://github.com/quantumlib/Cirq/pull/3871",
  "diff_url": "https://github.com/quantumlib/Cirq/pull/3871.diff",
  "patch_url": "https://github.com/quantumlib/Cirq/pull/3871.patch",
  "issue_url": "https://api.github.com/repos/quantumlib/Cirq/issues/3871",
  "number": 3871,
  "state": "open",
  "locked": false,
  "title": "[XEB] Sundry practical improvements",
  "user": {
    "login": "mpharrigan",
    "id": 4967059,
    "node_id": "MDQ6VXNlcjQ5NjcwNTk=",
    "avatar_url": "https://avatars.githubusercontent.com/u/4967059?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/mpharrigan",
    "html_url": "https://github.com/mpharrigan",
    "followers_url": "https://api.github.com/users/mpharrigan/followers",
    "following_url": "https://api.github.com/users/mpharrigan/following{/other_user}",
    "gists_url": "https://api.github.com/users/mpharrigan/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/mpharrigan/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/mpharrigan/subscriptions",
    "organizations_url": "https://api.github.com/users/mpharrigan/orgs",
    "repos_url": "https://api.github.com/users/mpharrigan/repos",
    "events_url": "https://api.github.com/users/mpharrigan/events{/privacy}",
    "received_events_url": "https://api.github.com/users/mpharrigan/received_events",
    "type": "User",
    "site_admin": false
  },
  "body": " - Shuffle order of task execution\r\n - Save data as you generate it\r\n - Move more functionality up the xxOptions Hierarchy\r\n - More robust fitting\r\n - Return std.dev. of fit parameters\r\n - Bugfix in notebook when running on device\r\n - Allow benchmarking the same pair in different layers and keep them distinct\r\n - Add note about characterizing the same pair in different layers: they will not be kept distinct. This can be relaxed, but requires changing the datatype of XEBCalibrationResult and keeping around (layer_i, pair_i) indices which are likely more troublesome to work with than actual qubit pairs. ",
  "created_at": "2021-03-01T19:47:04Z",
  "updated_at": "2021-03-05T00:26:57Z",
  "closed_at": null,
  "merged_at": null,
  "merge_commit_sha": "3abbe7313197e1d00c23cd66e9e306d65a09bd35",
  "assignee": null,
  "assignees": [

  ],
  "requested_reviewers": [
    {
      "login": "cduck",
      "id": 2476062,
      "node_id": "MDQ6VXNlcjI0NzYwNjI=",
      "avatar_url": "https://avatars.githubusercontent.com/u/2476062?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/cduck",
      "html_url": "https://github.com/cduck",
      "followers_url": "https://api.github.com/users/cduck/followers",
      "following_url": "https://api.github.com/users/cduck/following{/other_user}",
      "gists_url": "https://api.github.com/users/cduck/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/cduck/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/cduck/subscriptions",
      "organizations_url": "https://api.github.com/users/cduck/orgs",
      "repos_url": "https://api.github.com/users/cduck/repos",
      "events_url": "https://api.github.com/users/cduck/events{/privacy}",
      "received_events_url": "https://api.github.com/users/cduck/received_events",
      "type": "User",
      "site_admin": false
    },
    {
      "login": "vtomole",
      "id": 8405160,
      "node_id": "MDQ6VXNlcjg0MDUxNjA=",
      "avatar_url": "https://avatars.githubusercontent.com/u/8405160?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/vtomole",
      "html_url": "https://github.com/vtomole",
      "followers_url": "https://api.github.com/users/vtomole/followers",
      "following_url": "https://api.github.com/users/vtomole/following{/other_user}",
      "gists_url": "https://api.github.com/users/vtomole/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/vtomole/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/vtomole/subscriptions",
      "organizations_url": "https://api.github.com/users/vtomole/orgs",
      "repos_url": "https://api.github.com/users/vtomole/repos",
      "events_url": "https://api.github.com/users/vtomole/events{/privacy}",
      "received_events_url": "https://api.github.com/users/vtomole/received_events",
      "type": "User",
      "site_admin": false
    }
  ],
  "requested_teams": [

  ],
  "labels": [
    {
      "id": 2360761064,
      "node_id": "MDU6TGFiZWwyMzYwNzYxMDY0",
      "url": "https://api.github.com/repos/quantumlib/Cirq/labels/area/xeb",
      "name": "area/xeb",
      "color": "1d76db",
      "default": false,
      "description": ""
    },
    {
      "id": 922112536,
      "node_id": "MDU6TGFiZWw5MjIxMTI1MzY=",
      "url": "https://api.github.com/repos/quantumlib/Cirq/labels/cla:%20yes",
      "name": "cla: yes",
      "color": "FFFFFF",
      "default": false,
      "description": "Makes googlebot stop complaining."
    }
  ],
  "milestone": null,
  "draft": false,
  "commits_url": "https://api.github.com/repos/quantumlib/Cirq/pulls/3871/commits",
  "review_comments_url": "https://api.github.com/repos/quantumlib/Cirq/pulls/3871/comments",
  "review_comment_url": "https://api.github.com/repos/quantumlib/Cirq/pulls/comments{/number}",
  "comments_url": "https://api.github.com/repos/quantumlib/Cirq/issues/3871/comments",
  "statuses_url": "https://api.github.com/repos/quantumlib/Cirq/statuses/f7a9743d83e52db8e8bccc905fe6500ee764fa68",
  "head": {
    "label": "mpharrigan:2021-03-xeb-6",
    "ref": "2021-03-xeb-6",
    "sha": "f7a9743d83e52db8e8bccc905fe6500ee764fa68",
    "user": {
      "login": "mpharrigan",
      "id": 4967059,
      "node_id": "MDQ6VXNlcjQ5NjcwNTk=",
      "avatar_url": "https://avatars.githubusercontent.com/u/4967059?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/mpharrigan",
      "html_url": "https://github.com/mpharrigan",
      "followers_url": "https://api.github.com/users/mpharrigan/followers",
      "following_url": "https://api.github.com/users/mpharrigan/following{/other_user}",
      "gists_url": "https://api.github.com/users/mpharrigan/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/mpharrigan/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/mpharrigan/subscriptions",
      "organizations_url": "https://api.github.com/users/mpharrigan/orgs",
      "repos_url": "https://api.github.com/users/mpharrigan/repos",
      "events_url": "https://api.github.com/users/mpharrigan/events{/privacy}",
      "received_events_url": "https://api.github.com/users/mpharrigan/received_events",
      "type": "User",
      "site_admin": false
    },
    "repo": {
      "id": 191255763,
      "node_id": "MDEwOlJlcG9zaXRvcnkxOTEyNTU3NjM=",
      "name": "Cirq",
      "full_name": "mpharrigan/Cirq",
      "private": false,
      "owner": {
        "login": "mpharrigan",
        "id": 4967059,
        "node_id": "MDQ6VXNlcjQ5NjcwNTk=",
        "avatar_url": "https://avatars.githubusercontent.com/u/4967059?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/mpharrigan",
        "html_url": "https://github.com/mpharrigan",
        "followers_url": "https://api.github.com/users/mpharrigan/followers",
        "following_url": "https://api.github.com/users/mpharrigan/following{/other_user}",
        "gists_url": "https://api.github.com/users/mpharrigan/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/mpharrigan/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/mpharrigan/subscriptions",
        "organizations_url": "https://api.github.com/users/mpharrigan/orgs",
        "repos_url": "https://api.github.com/users/mpharrigan/repos",
        "events_url": "https://api.github.com/users/mpharrigan/events{/privacy}",
        "received_events_url": "https://api.github.com/users/mpharrigan/received_events",
        "type": "User",
        "site_admin": false
      },
      "html_url": "https://github.com/mpharrigan/Cirq",
      "description": "A python framework for creating, editing, and invoking Noisy Intermediate Scale Quantum (NISQ) circuits.",
      "fork": true,
      "url": "https://api.github.com/repos/mpharrigan/Cirq",
      "forks_url": "https://api.github.com/repos/mpharrigan/Cirq/forks",
      "keys_url": "https://api.github.com/repos/mpharrigan/Cirq/keys{/key_id}",
      "collaborators_url": "https://api.github.com/repos/mpharrigan/Cirq/collaborators{/collaborator}",
      "teams_url": "https://api.github.com/repos/mpharrigan/Cirq/teams",
      "hooks_url": "https://api.github.com/repos/mpharrigan/Cirq/hooks",
      "issue_events_url": "https://api.github.com/repos/mpharrigan/Cirq/issues/events{/number}",
      "events_url": "https://api.github.com/repos/mpharrigan/Cirq/events",
      "assignees_url": "https://api.github.com/repos/mpharrigan/Cirq/assignees{/user}",
      "branches_url": "https://api.github.com/repos/mpharrigan/Cirq/branches{/branch}",
      "tags_url": "https://api.github.com/repos/mpharrigan/Cirq/tags",
      "blobs_url": "https://api.github.com/repos/mpharrigan/Cirq/git/blobs{/sha}",
      "git_tags_url": "https://api.github.com/repos/mpharrigan/Cirq/git/tags{/sha}",
      "git_refs_url": "https://api.github.com/repos/mpharrigan/Cirq/git/refs{/sha}",
      "trees_url": "https://api.github.com/repos/mpharrigan/Cirq/git/trees{/sha}",
      "statuses_url": "https://api.github.com/repos/mpharrigan/Cirq/statuses/{sha}",
      "languages_url": "https://api.github.com/repos/mpharrigan/Cirq/languages",
      "stargazers_url": "https://api.github.com/repos/mpharrigan/Cirq/stargazers",
      "contributors_url": "https://api.github.com/repos/mpharrigan/Cirq/contributors",
      "subscribers_url": "https://api.github.com/repos/mpharrigan/Cirq/subscribers",
      "subscription_url": "https://api.github.com/repos/mpharrigan/Cirq/subscription",
      "commits_url": "https://api.github.com/repos/mpharrigan/Cirq/commits{/sha}",
      "git_commits_url": "https://api.github.com/repos/mpharrigan/Cirq/git/commits{/sha}",
      "comments_url": "https://api.github.com/repos/mpharrigan/Cirq/comments{/number}",
      "issue_comment_url": "https://api.github.com/repos/mpharrigan/Cirq/issues/comments{/number}",
      "contents_url": "https://api.github.com/repos/mpharrigan/Cirq/contents/{+path}",
      "compare_url": "https://api.github.com/repos/mpharrigan/Cirq/compare/{base}...{head}",
      "merges_url": "https://api.github.com/repos/mpharrigan/Cirq/merges",
      "archive_url": "https://api.github.com/repos/mpharrigan/Cirq/{archive_format}{/ref}",
      "downloads_url": "https://api.github.com/repos/mpharrigan/Cirq/downloads",
      "issues_url": "https://api.github.com/repos/mpharrigan/Cirq/issues{/number}",
      "pulls_url": "https://api.github.com/repos/mpharrigan/Cirq/pulls{/number}",
      "milestones_url": "https://api.github.com/repos/mpharrigan/Cirq/milestones{/number}",
      "notifications_url": "https://api.github.com/repos/mpharrigan/Cirq/notifications{?since,all,participating}",
      "labels_url": "https://api.github.com/repos/mpharrigan/Cirq/labels{/name}",
      "releases_url": "https://api.github.com/repos/mpharrigan/Cirq/releases{/id}",
      "deployments_url": "https://api.github.com/repos/mpharrigan/Cirq/deployments",
      "created_at": "2019-06-10T22:44:18Z",
      "updated_at": "2021-02-04T00:32:09Z",
      "pushed_at": "2021-03-04T23:47:42Z",
      "git_url": "git://github.com/mpharrigan/Cirq.git",
      "ssh_url": "git@github.com:mpharrigan/Cirq.git",
      "clone_url": "https://github.com/mpharrigan/Cirq.git",
      "svn_url": "https://github.com/mpharrigan/Cirq",
      "homepage": "",
      "size": 15558,
      "stargazers_count": 0,
      "watchers_count": 0,
      "language": "Python",
      "has_issues": false,
      "has_projects": true,
      "has_downloads": true,
      "has_wiki": true,
      "has_pages": false,
      "forks_count": 0,
      "mirror_url": null,
      "archived": false,
      "disabled": false,
      "open_issues_count": 8,
      "license": {
        "key": "apache-2.0",
        "name": "Apache License 2.0",
        "spdx_id": "Apache-2.0",
        "url": "https://api.github.com/licenses/apache-2.0",
        "node_id": "MDc6TGljZW5zZTI="
      },
      "forks": 0,
      "open_issues": 8,
      "watchers": 0,
      "default_branch": "master"
    }
  },
  "base": {
    "label": "quantumlib:master",
    "ref": "master",
    "sha": "af6500e468ea804f7e95f0e323770161391510a3",
    "user": {
      "login": "quantumlib",
      "id": 31279789,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjMxMjc5Nzg5",
      "avatar_url": "https://avatars.githubusercontent.com/u/31279789?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/quantumlib",
      "html_url": "https://github.com/quantumlib",
      "followers_url": "https://api.github.com/users/quantumlib/followers",
      "following_url": "https://api.github.com/users/quantumlib/following{/other_user}",
      "gists_url": "https://api.github.com/users/quantumlib/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/quantumlib/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/quantumlib/subscriptions",
      "organizations_url": "https://api.github.com/users/quantumlib/orgs",
      "repos_url": "https://api.github.com/users/quantumlib/repos",
      "events_url": "https://api.github.com/users/quantumlib/events{/privacy}",
      "received_events_url": "https://api.github.com/users/quantumlib/received_events",
      "type": "Organization",
      "site_admin": false
    },
    "repo": {
      "id": 114306758,
      "node_id": "MDEwOlJlcG9zaXRvcnkxMTQzMDY3NTg=",
      "name": "Cirq",
      "full_name": "quantumlib/Cirq",
      "private": false,
      "owner": {
        "login": "quantumlib",
        "id": 31279789,
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjMxMjc5Nzg5",
        "avatar_url": "https://avatars.githubusercontent.com/u/31279789?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/quantumlib",
        "html_url": "https://github.com/quantumlib",
        "followers_url": "https://api.github.com/users/quantumlib/followers",
        "following_url": "https://api.github.com/users/quantumlib/following{/other_user}",
        "gists_url": "https://api.github.com/users/quantumlib/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/quantumlib/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/quantumlib/subscriptions",
        "organizations_url": "https://api.github.com/users/quantumlib/orgs",
        "repos_url": "https://api.github.com/users/quantumlib/repos",
        "events_url": "https://api.github.com/users/quantumlib/events{/privacy}",
        "received_events_url": "https://api.github.com/users/quantumlib/received_events",
        "type": "Organization",
        "site_admin": false
      },
      "html_url": "https://github.com/quantumlib/Cirq",
      "description": "A python framework for creating, editing, and invoking Noisy Intermediate Scale Quantum (NISQ) circuits.",
      "fork": false,
      "url": "https://api.github.com/repos/quantumlib/Cirq",
      "forks_url": "https://api.github.com/repos/quantumlib/Cirq/forks",
      "keys_url": "https://api.github.com/repos/quantumlib/Cirq/keys{/key_id}",
      "collaborators_url": "https://api.github.com/repos/quantumlib/Cirq/collaborators{/collaborator}",
      "teams_url": "https://api.github.com/repos/quantumlib/Cirq/teams",
      "hooks_url": "https://api.github.com/repos/quantumlib/Cirq/hooks",
      "issue_events_url": "https://api.github.com/repos/quantumlib/Cirq/issues/events{/number}",
      "events_url": "https://api.github.com/repos/quantumlib/Cirq/events",
      "assignees_url": "https://api.github.com/repos/quantumlib/Cirq/assignees{/user}",
      "branches_url": "https://api.github.com/repos/quantumlib/Cirq/branches{/branch}",
      "tags_url": "https://api.github.com/repos/quantumlib/Cirq/tags",
      "blobs_url": "https://api.github.com/repos/quantumlib/Cirq/git/blobs{/sha}",
      "git_tags_url": "https://api.github.com/repos/quantumlib/Cirq/git/tags{/sha}",
      "git_refs_url": "https://api.github.com/repos/quantumlib/Cirq/git/refs{/sha}",
      "trees_url": "https://api.github.com/repos/quantumlib/Cirq/git/trees{/sha}",
      "statuses_url": "https://api.github.com/repos/quantumlib/Cirq/statuses/{sha}",
      "languages_url": "https://api.github.com/repos/quantumlib/Cirq/languages",
      "stargazers_url": "https://api.github.com/repos/quantumlib/Cirq/stargazers",
      "contributors_url": "https://api.github.com/repos/quantumlib/Cirq/contributors",
      "subscribers_url": "https://api.github.com/repos/quantumlib/Cirq/subscribers",
      "subscription_url": "https://api.github.com/repos/quantumlib/Cirq/subscription",
      "commits_url": "https://api.github.com/repos/quantumlib/Cirq/commits{/sha}",
      "git_commits_url": "https://api.github.com/repos/quantumlib/Cirq/git/commits{/sha}",
      "comments_url": "https://api.github.com/repos/quantumlib/Cirq/comments{/number}",
      "issue_comment_url": "https://api.github.com/repos/quantumlib/Cirq/issues/comments{/number}",
      "contents_url": "https://api.github.com/repos/quantumlib/Cirq/contents/{+path}",
      "compare_url": "https://api.github.com/repos/quantumlib/Cirq/compare/{base}...{head}",
      "merges_url": "https://api.github.com/repos/quantumlib/Cirq/merges",
      "archive_url": "https://api.github.com/repos/quantumlib/Cirq/{archive_format}{/ref}",
      "downloads_url": "https://api.github.com/repos/quantumlib/Cirq/downloads",
      "issues_url": "https://api.github.com/repos/quantumlib/Cirq/issues{/number}",
      "pulls_url": "https://api.github.com/repos/quantumlib/Cirq/pulls{/number}",
      "milestones_url": "https://api.github.com/repos/quantumlib/Cirq/milestones{/number}",
      "notifications_url": "https://api.github.com/repos/quantumlib/Cirq/notifications{?since,all,participating}",
      "labels_url": "https://api.github.com/repos/quantumlib/Cirq/labels{/name}",
      "releases_url": "https://api.github.com/repos/quantumlib/Cirq/releases{/id}",
      "deployments_url": "https://api.github.com/repos/quantumlib/Cirq/deployments",
      "created_at": "2017-12-14T23:41:49Z",
      "updated_at": "2021-03-04T23:43:28Z",
      "pushed_at": "2021-03-05T00:24:31Z",
      "git_url": "git://github.com/quantumlib/Cirq.git",
      "ssh_url": "git@github.com:quantumlib/Cirq.git",
      "clone_url": "https://github.com/quantumlib/Cirq.git",
      "svn_url": "https://github.com/quantumlib/Cirq",
      "homepage": "",
      "size": 15140,
      "stargazers_count": 2743,
      "watchers_count": 2743,
      "language": "Python",
      "has_issues": true,
      "has_projects": true,
      "has_downloads": true,
      "has_wiki": true,
      "has_pages": false,
      "forks_count": 576,
      "mirror_url": null,
      "archived": false,
      "disabled": false,
      "open_issues_count": 364,
      "license": {
        "key": "apache-2.0",
        "name": "Apache License 2.0",
        "spdx_id": "Apache-2.0",
        "url": "https://api.github.com/licenses/apache-2.0",
        "node_id": "MDc6TGljZW5zZTI="
      },
      "forks": 576,
      "open_issues": 364,
      "watchers": 2743,
      "default_branch": "master"
    }
  },
  "_links": {
    "self": {
      "href": "https://api.github.com/repos/quantumlib/Cirq/pulls/3871"
    },
    "html": {
      "href": "https://github.com/quantumlib/Cirq/pull/3871"
    },
    "issue": {
      "href": "https://api.github.com/repos/quantumlib/Cirq/issues/3871"
    },
    "comments": {
      "href": "https://api.github.com/repos/quantumlib/Cirq/issues/3871/comments"
    },
    "review_comments": {
      "href": "https://api.github.com/repos/quantumlib/Cirq/pulls/3871/comments"
    },
    "review_comment": {
      "href": "https://api.github.com/repos/quantumlib/Cirq/pulls/comments{/number}"
    },
    "commits": {
      "href": "https://api.github.com/repos/quantumlib/Cirq/pulls/3871/commits"
    },
    "statuses": {
      "href": "https://api.github.com/repos/quantumlib/Cirq/statuses/f7a9743d83e52db8e8bccc905fe6500ee764fa68"
    }
  },
  "author_association": "COLLABORATOR",
  "auto_merge": null,
  "active_lock_reason": null,
  "merged": false,
  "mergeable": true,
  "rebaseable": true,
  "mergeable_state": "clean",
  "merged_by": null,
  "comments": 5,
  "review_comments": 6,
  "maintainer_can_modify": true,
  "commits": 10,
  "additions": 137,
  "deletions": 37,
  "changed_files": 5
}
```

</details>

**Pull request merged**